pkg/scheduler/framework/plugins/dynamicresources/dynamicresources_test.go

```diff
@@ -197,6 +198,102 @@ type prepare struct {
 	postfilter change
 }
 
+func BenchmarkHaveAllNodes(b *testing.B) {
+	tests := []*struct {
+		name      string
+		nodeNames []string
+		nodes     []*v1.Node
+		nodeCount int
+	}{
+		{
+			name:      "have-nodes-10",
+			nodeCount: 10,
+		},
+		{
+			name:      "have-nodes-50",
+			nodeCount: 50,
+		},
+		{
+			name:      "have-nodes-100",
+			nodeCount: 100,
+		},
+		{
+			name:      "have-nodes-500",
+			nodeCount: 500,
+		},
+	}
+	for _, test := range tests {
+		for i := 0; i < test.nodeCount; i++ {
+			test.nodeNames = append(test.nodeNames, "worker-"+strconv.Itoa(i))
+			test.nodes = append(test.nodes, &st.MakeNode().Name("worker-"+strconv.Itoa(i)).Node)
+		}
+		b.Run(test.name, func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				haveAllNodes(test.nodeNames, test.nodes)
+			}
+		})
+	}
+}
+
+func TestHaveAllNodes(t *testing.T) {
+	tests := []*struct {
+		name      string
+		nodeNames []string
+		nodes     []*v1.Node
+		has       bool
+	}{
+		{
+			name:      "same-nodes",
+			nodeNames: []string{"worker-a"},
+			nodes: []*v1.Node{
+				&st.MakeNode().Name("worker-a").Node,
+			},
+			has: true,
+		},
+		{
+			name:      "have-all",
+			nodeNames: []string{"worker-a", "worker-b"},
+			nodes: []*v1.Node{
+				&st.MakeNode().Name("worker-a").Node,
+			},
+			has: true,
+		},
+		{
+			name:      "not-all",
+			nodeNames: []string{"worker-a"},
+			nodes: []*v1.Node{
+				&st.MakeNode().Name("worker-a").Node,
+				&st.MakeNode().Name("worker-b").Node,
+			},
+			has: false,
+		},
+	}
+	// Test for the time complexity is O(n*n)
+	for _, test := range tests {
+		for i := 0; i < 40; i++ {
```

Comment:
Why not the `haveAllNodesThreshold` but 40?

Reply:
I think you can move this out of the for-loop for performance.

Reply:
done. PTAL, thanks.